### PR TITLE
Database backend abstractions for status

### PR DESCRIPTION
**Pull request by eddowh**
Ran `flake8` against every comitted file; passed more or less.

Have not tested with dev machine yet; left a few `XXX` notes here and there as questions

Covers #42

### CODE REVIEW

File: openkongqi/conf.py [hunk 1]
```diff
@@ -132,6 +141,13 @@ def load_logging():
 # load sources from sources directory
 settings['SOURCES'] = load_tree(settings['SOURCES_DIR'], dig_loader)
 
+
+def create_db_instance(database):
+    """Create database instance given a database settings dict."""
+    db_mod = load_backend(database['ENGINE'])
+    return db_mod.StatusWrapper(database)
+settings['DATABASE'] = create_db_instance(settings['DATABASES']['status'])
```
Comment (hrbonz) [inline openkongqi/conf.py]: Why save it in the settings dict? it's not a settings entry.
Just create a `statusdb` variable in the module so I can `from settings import statusdb`

File: openkongqi/status/base.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class BaseStatusWrapper:
+    """Base wrapper class to get database status. This class is to be used
+    as parent of any database status wrapper class.
+
+    The children wrapper classes will have to define custom methods for
+    creating the database connection.
+    """
+    def __init__(self, db_settings):
+        self._cnx = self.create_cnx(db_settings)
```
Comment (hrbonz) [inline openkongqi/status/base.py]: Where's the prototype for `self.create_cnx()`?

File: openkongqi/status/redis.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    def create_cnx(self, db_settings):
+        """Create a connection to database.
+
+        :param db_settings: dictionary containing database configuration
+        :type db_settings: dict
+        """
+        # keep everything but the 'ENGINE' key
+        needed_kwargs = db_settings.keys()
+        needed_kwargs.remove('ENGINE')
+        db_settings_kwargs = {
```
Comment (hrbonz) [inline openkongqi/status/redis.py]: I'd rather see something with defaults here. Create some default for host, db_id, etc at the module level and for the settings you can do something similar to `db_settings.get('HOST', _HOST)`

File: openkongqi/status/base.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+
+STATUS_KEY = "okq_status:{name}"
```
Comment (hrbonz) [inline openkongqi/status/base.py]: The `STATUS_KEY` is very specific to redis, it should not live in this module.

File: openkongqi/utils.py [hunk 5]
```diff
@@ -46,6 +47,21 @@ def dig_loader(base_uuid, data):
...
+
+    :param backend_name: database backend to use
+    :type backend_name: str
+    :returns: module
+    """
+    try:
+        return import_module(backend_name)
+    except ImportError as e:
+        # FIXME add error treatment
+        # XXX - why not just let the import fail?
```
Comment (hrbonz) [inline openkongqi/utils.py]: Remove this when pushing your branch please, this discussion should happen on the PR.

File: openkongqi/status/base.py (hunk 2 above)
Comment (eddowh) [inline openkongqi/status/base.py]: What do you mean by "prototype"?

**Pull request merged**